src/envelope.rs

```diff
@@ -25,12 +25,9 @@ impl<A, M> Envelope<A> for SendEnvelope<M>
 {
 	fn handle( self: Box<Self>, actor: &mut A ) -> Return<()>
 	{
-		Box::pin( async move
-		{
-
-			let _ = < A as Handler<M> >::handle( actor, self.msg ).await;
-
-		})
+		< A as Handler<M>  >::handle( actor, self.msg )
+			.map(|_| ())
+			.boxed()
```

Comment:
@njam Can you explain to me why this solves the `Send` requirement? Since `handle` is async and it takes a `&mut self`, I would think that a reference to the actor would at least be in the future produced by `handle`, making me think Actor should at least be `Sync` here?

Somehow I'm missing the obvious...

Reply:
Hmm, I realize why this is. It's because the `handle` method specifies that the future it returns is `Send`, which means that it's enforced in user code rather than here. That is a problem, because it means that a `!Send` actor cannot use `self` in the future `handle` returns. 

I will try to solve this, but I'm not sure how yet.

Reply:
Took me a while to understand. So the future returned from `handle()` must be `Send`, so that the Future of the inbox that is processing all messages can be `Send` too, which in turn is required, so that the inbox can be spawned on a separate thread.

Conceptually this seems weird, because we just want to transfer the inbox Future to another thread, but any values returned from `handle()` will _stay_ on that thread and be transferred to other threads through a channel. But I guess since the futures are combined into a single big future, everything must be `Send`, or nothing?  

Maybe an alternative would be to spawn the inbox future on another thread, and _in_ that inbox future start _another_ future executor to call `handle()` on the actor. That way the returned future of `handle()` would never have to be `Send`.

Reply:
I found it confusion as well, but it turns out, a future cannot be `Send` if it awaits any future that isn't `Send`. So even though it awaits rather than spawns the future from `handle`, it still has to be `Send`. The I think this property is used by executors to be able to move a task from one thread to another (work stealing) even after it has started executing. 

It's unfortunate to have to double out the methods, but it turned out that here the impact on the user facing API remained small, and double code is limited to a few tens of lines, so it's well worth it.

While working on the remote actors in WASM, I still found limitations. The `Peer` actor  would best be `Send` so users can spawn connection handling on a thread pool, but it is generic on a user supplied type of connection. In WASM, about nothing is `Send`. And WASM doesn't even have threads right now, but if I want to reuse the code, the `Send` bound becomes a problem. Since here it's a library that provides the actor, but client code provides a generic type, which determines the `Send`-ness. For application actors this is not so much a problem, but for library the eternal problem is being obliged to define `Send`-ness, when it should depend on client types.

I will find a work-around, and thespis documentation shall have to guide users to solve this kind of problems.